### PR TITLE
backend service development workshop

### PR DESCRIPTION
**Pull request by arbrix**
I

### CODE REVIEW

File: calendar/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+Add persistence layer. Now all information about events and user should be stored inside database.
+
+### TODO
+
+* Organize DB connection (learn what parameters it accept and what of them should be used), make it configurable; (use Docker to start PostgreSQL)
+* Use DB migration to setup schema on the start;
+* Update logic to store data in the DB;
+* Add integration test to cover interation with a DB; 
+
+### Hint:
```
Comment (Metalscreame) [inline calendar/README.md:336]:
```
### 5. Add gRPC:
   * define protobuff for your task and user models
   * define protobuff definition for your Task APIs  
   * use protoc tool to generate grpc stubs
   * define auth interceptor (gRPC middleware) that will reuse same logic from http auth md
   * add second main file that will serve only gRPC requests
   * write gRPC client that will call one of your APIs
   * install BloomRPC (postman analog for gRPC) to test your APIs

Links:

    https://developers.google.com/protocol-buffers/docs/gotutorial
    https://grpc.io/docs/languages/go/basics/
   ```

File: calendar/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    
+* GET events (with filter by date/data-range, time range, title), 
+* GET event by ID, 
+* POST create an event, 
+* PUT update an event (title, description, data & time, timezone, duration, notes), 
+* DELETE delete event.
+
+### Hints
+
+* Start from utilizing standard packages: `net/http`;
```
Comment (anastasiia-skliar) [inline calendar/README.md]: https://quii.gitbook.io/learn-go-with-tests/build-an-application/app-intro
tutorial for simple http server with tests

Comment (arbrix) [reply calendar/README.md]: Thank you, added

File: calendar/README.md (hunk 1 above)
Comment (arbrix) [reply calendar/README.md:336]: Great idea, added as well.

File: calendar/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+### REST CRUD for events;
+    
+* GET events (with filter by date/data-range, time range, title), 
+* GET event by ID, 
+* POST create an event, 
+* PUT update an event (title, description, data & time, timezone, duration, notes), 
+* DELETE delete event.
+
+### Hints
```
Comment (anastasiia-skliar) [inline calendar/README.md:245]: https://play.golang.org/p/l_32IxixTMK

Example how to work with timezones

File: calendar/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+## 2. Serve multiple users
+
+Store events per user. For this you require to have authentication/authorization; User should have own time zone and all events should respect it by default.
+
+### TODO
+
+* Add login/logout enpoints (credantials should be stored in memory as well);
+* Use JWT for authorization at other endpoints;
+* Update logic to Events endpoints to authorize access only to related to user events.
+* PUT for update user timezone; Time of events should be returned in a new timezone.
```
Comment (anastasiia-skliar) [inline calendar/README.md:283]: Add swagger ?

File: calendar/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+### Hints
+
+* Start from utilizing standard packages: `net/http`; 
+  * [tutorial for simple http server with tests](https://quii.gitbook.io/learn-go-with-tests/build-an-application/app-intro)
+  * [http handlers](https://quii.gitbook.io/learn-go-with-tests/questions-and-answers/http-handlers-revisited)
+  * [repo with code from learn go with tests](https://github.com/quii/learn-go-with-tests)
+* Use `net/http/httptest` for creation tests;
+* Fineout how to convert & store time between different timezones with package `time`;
+* All data should be stored in memory;
+* Use Postman to validate your server;
```
Comment (Metalscreame) [inline calendar/README.md:254]: - use go mod 
- use gomock || testify  for testing

**Pull request merged**